iml-gui/crate/src/page/snapshot/create_policy.rs

```diff
@@ -0,0 +1,437 @@
+    pub modal: modal::Model,
+    submitting: bool,
+    filesystems: Vec<String>,
+    policies: HashMap<String, Arc<SnapshotPolicy>>,
+    interval: u64,
+    interval_unit: String,
+    vars: snapshot::policy::create::Vars,
+}
+
+impl Model {
+    fn refresh(&mut self, cache: &ArcCache) {
+        self.filesystems = cache.filesystem.iter().map(|(_, x)| x.name.clone()).collect();
+        self.policies = cache
+            .snapshot_policy
+            .iter()
+            .map(|(_, x)| (x.filesystem.clone(), x.clone()))
+            .collect();
+    }
+}
+
+impl RecordChange<Msg> for Model {
+    fn update_record(&mut self, _: ArcRecord, cache: &ArcCache, orders: &mut impl Orders<Msg, GMsg>) {
+        self.refresh(cache);
+    }
+    fn remove_record(&mut self, _: RecordId, cache: &ArcCache, orders: &mut impl Orders<Msg, GMsg>) {
+        self.refresh(cache);
+
+        let present = cache.filesystem.values().any(|x| x.name == self.vars.filesystem);
+
+        if !present {
+            let x = get_fs_names(cache).into_iter().next().unwrap_or_default();
+            orders.send_msg(Msg::Input(Input::Filesystem(x)));
+        }
+    }
+    fn set_records(&mut self, cache: &ArcCache, orders: &mut impl Orders<Msg, GMsg>) {
+        self.refresh(cache);
+
+        let x = get_fs_names(cache).into_iter().next().unwrap_or_default();
+        orders.send_msg(Msg::Input(Input::Filesystem(x)));
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum Msg {
+    Modal(modal::Msg),
+    Open,
+    Close,
+    Input(Input),
+    Submit,
+    CreatePolicyResp(fetch::ResponseDataResult<Response<snapshot::policy::create::Resp>>),
+    Noop,
+}
+
+#[derive(Clone, Debug)]
+pub enum Input {
+    Filesystem(String),
+    Interval(u64),
+    IntervalUnit(String),
+    ToggleBarrier,
+    Keep(i32),
+    Daily(Option<i32>),
+    Monthly(Option<i32>),
+    Weekly(Option<i32>),
+}
+
+pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg, GMsg>) {
+    match msg {
+        Msg::Input(x) => match x {
+            Input::Interval(i) => model.interval = i,
+            Input::IntervalUnit(i) => model.interval_unit = i,
+
+            Input::Filesystem(i) => {
+                if let Some(p) = model.policies.get(&i) {
+                    model.vars.keep = p.keep;
+                    model.vars.daily = Some(p.daily);
+                    model.vars.weekly = Some(p.weekly);
+                    model.vars.monthly = Some(p.monthly);
+                    model.vars.barrier = Some(p.barrier);
+                    model.interval = p.interval.0.as_secs() / 60;
+                    model.interval_unit = "m".into();
+                } else {
+                    model.vars = Default::default();
+                    model.interval = 60;
+                    model.interval_unit = "m".into();
+                }
+                model.vars.filesystem = i;
+            }
+            Input::ToggleBarrier => model.vars.barrier = Some(!model.vars.barrier.unwrap_or(false)),
```

Comment:
```suggestion
            Input::ToggleBarrier => model.vars.barrier = Some(!model.vars.barrier.unwrap_or_default()),
```

Is a bit more idiomatic I think

Reply:
Explicit is better than implicit :)